task_reactredux/task1/src/containers/CounterContainer.js

```diff
@@ -31,14 +32,58 @@ class CounterContainer extends React.Component {
         });
     }
 
+    componentDidMount(){
+        console.log("Child: componentDidMount()");
+    }
+
+    componentDidUpdate(){
+        console.log("Child: componentDidUpdate()");
+    }
+
+    componentWillUnmount(){
+        console.log("Child: componentWillUnmount()");
+    }
+
+    UNSAFE_componentWillReceiveProps(prop) {
+        console.log("Child: componentWillReceiveProps()");
+
+        let counter = this.state.count;
+
+        if ((prop.command === commandsEnum.add) && (counter % 2 === 0) && (counter !== 0)) {
```

Comment:
Probably it's better to move `(counter % 2 === 0)` check into function, smth like `checkNumberOnPatiry`

Reply:
Also, do you need additional braces? `if (prop.command === commandsEnum.add && counter % 2 === 0 && counter !== 0) {`